### ISSUE TITLE
Valley Bottom Segmentation Tool

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/vbet#7]
To get useful metrics and measures out of VBET (and dashboard summaries), we need the valley bottom to be something other than a liklihood raster (#5) or a single polygon blob (#6). I am going to call this **Valley Bottom Segmentation**. It is reasonably illustrated below (from these [slides](https://restoration-usu.github.io/Ecohydraulics/Modules/NetworkTools.html#network-scale-assessments)):
![image](https://user-images.githubusercontent.com/9723905/80763292-72764780-8afb-11ea-8288-24ac2627ba4c.png)

A valley bottom segment could be defined as a longitudinal portion (reach or segment) of a riverscape, which spans the entire valley bottom width. They will typically be longer (longitudinally) than it is laterally (cross-valley). 

# Ways to get them
In a [commercial-grade](riverscapes.xyz/Tools/#tool-grade) or [professional-grade](riverscapes.xyz/Tools/#tool-grade) version of VBET,  we might let the user choose which way to get there, or let them choose the valley bottom segmentation method and produce multiple **valley bottom segmentation** realizations.  These include (not mutually-exclusive... could be combined or used to further split/segment):
- Load an already segmented network, segments drawn perpendicularity from all reach segment end points into VB intersection. 
- Segment based on tributary (of intput VBET network) junctions - *Note: this would be really important that the end  of the tributary includes the fan of that tributary.*
- Further segment based on intersection with polygon input (e.g. of land ownership or property boundaries) - pyBRAT has this option
- Further segment based on intersection of polyline input with drainage network (e.g. road crossings) - *Note: it is important not to allow this for polyline intersection of VBET polygon, because many roads run parallel.*
- Further segment based on a uniform segmentation length (e.g. RCAT uses 500 m and BRAT uses 300 m by default).
- Further segment based on variable segmentation length defined by continuous network function (this is what Fluvial Coridor Toolbox can do and GNAT implemented as well).

# Why the above are important
The valley bottom segment is important as it defines the length-scale over-which potential valley bottom derivatives and metrics can be summarized. 

@philipbaileynar - this is one way we give users ownership of their own VBETs.

**Comment by philipbaileynar** [comment]
I appreciate the value of this and we can make progress here, but this is going to require lots of 
 discussion. Everything looks so perfect and tidy in an isolated single channel situation. But the image below is the NHD network (red) in Bear Valley (16010202) with 60% VBET in green. Where's the valley? What's a tributary?

![Screen Shot 2020-04-30 at 4 01 19 PM](https://user-images.githubusercontent.com/8645379/80767208-3d1e2980-8afc-11ea-8065-0008370b787b.png)

I guess my questions are:

* Is segmentation something we do everywhere or something we run on a case by case basis in places of interest/restoration/funding?
* The segments in your graphic are perpendicular to the valley centerline. This tells me we need to incorporate a centerline algorithm.
*  What is a tributary? Is it any junction? Or only where stream order changes? Below is a more sensible location in the same Bear Valley HUC. How many valleys and tribs are shown? (I know VBET needs to be extended for the entire network not just the perennial).

![Screen Shot 2020-04-30 at 4 10 46 PM](https://user-images.githubusercontent.com/8645379/80767622-27f5ca80-8afd-11ea-99c8-49f007783d60.png)

In summary... we can develop a segmentation, but I'm not sure it makes sense everywhere and I think generic rules that work everywhere are going to be difficult.
